ProcessMaker/processmaker
# FOUR-10750 Licensing Product Integration

Licensing Product Integration (Part 1)

## Solution
- Implement a Service Provider to enable disable packages.
- Implement an artisan command to update or install a license: `processmaker:license-update`
- Implement an artisan command to remove a license: `processmaker:license-remove`

## How to Test
- Create a json file (e.g. named license.json) with a content similar to:
```
{
  "expires_at": "2023-12-31",
  "packages": [
    "package-translations",
    "package-projects"
  ]
}
```
- Run the artisan command: `processmaker:license-update license.json`
- Only the packages translations and projects will be enabled (make sure you have instaled the packages you want to test)
- Run the artisan command: `processmaker:license-remove` all the installed packages will be enabled

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-10750
- https://processmaker.atlassian.net/browse/FOUR-11055
## Code Review Checklist

ci:next
